packages/nimble-components/src/patterns/error/specs/checkbox-radio-error.md

```diff
@@ -0,0 +1,65 @@
+# Error State on Checkbox and Radio Group
+
+## Overview
+
+The `nimble-checkbox` and the `nimble-radio-group` need to have error states associated with them. Following the existing patterns within nimble, this error state will be a red exclamation mark icon next to the component and red error text below the component.
+
+### Background
+
+-   [Visual design spec](https://www.figma.com/design/PO9mFOu5BCl8aJvFchEeuN/Nimble_Components?node-id=1295-63148&node-type=canvas&t=sJ2Fhm1vLoZ4zpsK-0)
+-   [Nimble issue 2018: Error state for nimble-checkbox](https://github.com/ni/nimble/issues/2018)
+-   [Nimble issue 2019: Error state for nimble-radio-group](https://github.com/ni/nimble/issues/2019)
+
+Related issues that can and/or should be resolved as part of this effort:
+
+-   [Nimble issue 2091: nimble-checkbox has incorrect height](https://github.com/ni/nimble/issues/2091)
+-   [Nimble issue 2090: Error icon is incorrectly spaced in nimble-text-field](https://github.com/ni/nimble/issues/2090)
+
+## Design
```

Comment:
What is the expectation/guidance for clients in terms of setting widths of checkboxes or radio groups? Currently, radio buttons in a group do not expand to the width of the group, so (in a vertical orientation) error icons would not line up unless the labels were all the same length. Will we automatically change `width` to `100%` when the button is within a group? Will we have guidance (or styling rules) to ensure the radio group width allows for some space between the longest radio button label and the error icon? Is it up to a client to ensure radio groups are wide enough that button labels never wrap? Otherwise, how does label wrapping affect icon placement?

What about radio groups with a horizontal orientation? Do we still put an error icon after each button label, and what spacing do we use?

For checkboxes, some of the same questions apply. Should the width default to `100%`? Is it okay if the client doesn't provide an error string, and the width of the checkbox results in huge separation between the icon and the label? E.g.
![image](https://github.com/user-attachments/assets/52ed0eb2-16c1-48e4-8eff-6c800714e7be)

What is the wrapping behavior of the error text?

Reply:
A lot of this information has now been incorporated into the HLD, but here are some answers:

> What is the expectation/guidance for clients in terms of setting widths of checkboxes or radio groups?

For the checkbox and vertical radio group, the expectation is that a client would set a width on it, just like the client would for any other control. I'm not entirely sure what the expectation is for a horizontal radio group. I don't believe we currently have any scenarios where that is the recommended orientation.

> Will we automatically change width to 100% when the button is within a group?

Yes, when the radio buttons are vertically oriented, they will be styled to have the same width as the group. I've updated the HLD.

> Will we have guidance (or styling rules) to ensure the radio group width allows for some space between the longest radio button label and the error icon?

The space between the label and the error icon will be part of the component's styling. The client won't have to do anything to accomplish this.

> Is it up to a client to ensure radio groups are wide enough that button labels never wrap? Otherwise, how does label wrapping affect icon placement?

The labels are allowed to wrap. The icon will always be center aligned with the first row of the text.

> What about radio groups with a horizontal orientation? Do we still put an error icon after each button label, and what spacing do we use?

There will still be an error icon after each button label using the same spacing. I think we need guidance in general about when to use the horizontal orientation of the radio buttons, but I don't think that has to be part of this effort of adding error states.

>What is the wrapping behavior of the error text?

As with the error text in other components, it does not wrap.